src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -264,6 +267,21 @@ def collect_extension_functions(
 
         return collected_extension_fns_by_path
 
+    def deannotate(
+        self, py_file: Path, extension_fns: List[NativeAppExtensionFunction]
+    ):
+        with open(py_file, "r", encoding="utf-8") as f:
+            code = f.read()
+
+        if py_file.is_symlink():
+            # if the file is a symlink, make sure we don't overwrite the original
+            py_file.unlink()
+
+        new_code = deannotate(code, extension_fns)
```

Comment:
nit: worth differentiating between extension_function_utils.deannotate vs this annotate? I understand that one way is to see the use of `self.deannotate` vs regular `deannotate`, but could help to differentiate for readability.